sympy/combinatorics/fp_groups.py

```diff
@@ -148,15 +150,76 @@ def order(self, strategy="relator_based"):
         2
 
         """
+        from sympy import S, gcd
         if self._order != None:
             return self._order
         if self._coset_table != None:
             self._order = len(self._coset_table.table)
+        elif len(self.generators) == 1:
+            self._order = gcd([r.array_form[0][1] for r in self.relators])
         else:
-            self._coset_table = self.coset_enumeration([], strategy)
-            self._order = len(self._coset_table.table)
+            gens, C = self._finite_index_subgroup()
+            if C:
+                ind = len(C.table)
+                self._order = ind*self.subgroup(gens, C=C).order()
+            else:
+                self._order = self.index([])
         return self._order
 
+    def _finite_index_subgroup(self, s=[]):
+        '''
+        Find the elements of `self` that generate a finite index subgroup
+        and return the list of elements and the coset table of `self` by
+        the subgroup
```

Comment:
Perhaps the docstring could be something like "Try to find elements ... the subgroup if found else return ...".